pkg/reconcile/reconcile.go

```diff
@@ -91,3 +93,25 @@ var _ Reconciler = Func(nil)
 
 // Reconcile implements Reconciler.
 func (r Func) Reconcile(o Request) (Result, error) { return r(o) }
+
+// PanicHandler can be called with defer in order to recover from panic in reconcile logic.
+// Additional handlers can also be passed to manage the go routines. The error passed to the
+// PanicHandler is set to the value of panic if not nil.
+// If any resource triggers a panic, the reconciler will recover and modify the raised panic
+// to the normal reconciler error.
+// Example usage:
+/*
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (result reconcile.Result, err error) {
```

Comment:
Very minor nit, but not sure about the name of the reconciler struct here `ReconcileMemcached`.
It might make sense in the context of Kubebuilder/Operator SDK scaffolds but maybe we can use the same example reconciler that's consistent with the existing example in the godocs for this pkg. 

https://github.com/kubernetes-sigs/controller-runtime/blob/c4893e0b992fb1e6e7da29acd692a0cdf9d7aec9/pkg/reconcile/example_test.go#L30-L34

Or I guess we have another one here:

https://github.com/kubernetes-sigs/controller-runtime/blob/c4893e0b992fb1e6e7da29acd692a0cdf9d7aec9/example_test.go#L119

Reply:
Agreed! Modified it to be the `ReplicaSetReconciler` as in `controller-runtime/example_test.go`